plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -82,6 +85,36 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Adds the product template ID to the product if it doesn't exist.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WC_Product       $product The product.
+	 */
+	public function possibly_add_template_id( $response, $product ) {
+		if ( ! $product ) {
+			return $response;
+		}
+		if ( ! $product->meta_exists( '_product_template_id' ) ) {
+			/**
+			 * Allows to determine a product template id based on the product data.
+			 *
+			 * @since 9.1.0
+			 */
```

Comment:
I think we should not document this filter for now, and mark it as experimental (I can't remember how we've done that for other hooks, but we should follow suit).